### ISSUE TITLE
Unhandled Exception for MyBikeTraffic in stationary testing

### ISSUE THREAD

**Issue by chrisoutwright** [kartoone/mybiketraffic#8]
I did a test indoors (Edge 840) with the datafield enabled and saw that after the first (albeit slow moving object simulated by a muffin mold plate that was pickup by Garmin Varia RCT715, and then dropped, as false positive) caused the field to drop the 0 Value and showed only the IQ logo there instead.

I see a entry CIQ_LOG.BAK

```
---
Error: Unhandled Exception
Time: 2024-02-18T15:17:58Z
Part-Number: 006-B4062-00
Firmware-Version: '19.22'
Language-Code: deu
ConnectIQ-Version: 4.2.3
Store-Id: c5d949c3-9acb-4e00-bb2d-c3b871e9e733
Store-Version: 65
Filename: E2IF0349
Appname: MyBikeTraffic
Stack: 
  - pc: 0x100016f4
  - pc: 0x10000234
  - 
```

Could this be when no GPS is picked up or speed cannot be determined as bike was stationary in testing?
I will be able to test outside when the raining stops. Probably on Tuesday.

**Comment by chrisoutwright** [comment]
As I understood, you can get via Error Reporting Application ([ERA](https://developer.garmin.com/connect-iq/core-topics/exception-reporting-tool/)) the info, is this correct?
I wonder where the Unhandled Exception happened, at the compute layer? Just a question:
bikeRadar.getRadarInfo() does it has fixed targets (size) of 8 always or could this be smaller in some cases?

**Comment by kartoone** [comment]
Yes, the app crashes when there is no sat connection ... I have specific checks in my code for this that should prevent it, but it crashes anyway. So I think this is actually a bug in the ConnectIQ firmware related to the BikeRadar object. The only thing I can think of to prevent this is to do a "satellite" check before calling line 115 in MyBikeTrafficFitContributions.mc ... I think this is what is crashing the app. What do you think?

**Comment by kartoone** [comment]
Update, I was able to open the crash log from ERA and discovered that this was a problem I fixed in December 2023, but the build that is on the Garmin store is still November 2023, so I either forgot to upload the build (highly doubt) or for some reason the new build didn't get approved. I have rebuilt and re-uploaded the app to fix the satellite connectivity crash. I think it takes a while for the new version to be approved and make it to the appstore. I will check back again in a few hours to see if the new version 2.7.3 has made it onto the store. Would you please recreate your exact same test to see if it still crashes once you have updated? Thanks.

**Comment by chrisoutwright** [comment]
Thanks for the quick answer!

I will test it as soon as I am back to my Edge device with the new version once it is out. 

A different Question on the counting system:
since compute() is called once a second, will the THRESH take into account the prior speed of the particular target?
Let's say it is 20 m away and THRESH is 10m, with more than 20m/sec, it would ignore the count (or what happens to range once that car is off the scan by pure speed and the 1 sec delay of compute ?: will it be saved for few secs still, till getting zero?) or would it determine that still? I had a look at where crossedthresh is set (also the line position before counts are made) and I was not getting it quite (did some simulation tests with simulated speeding cars).

**Comment by kartoone** [comment]
Thanks, new version should be out now. Let me know if you still get it to crash in your indoor test. Also, regarding the really fast car, unfortunately it would miss that car. It's hard to distinguish that car from a car that turned between 20 and 10m behind you (before crossing the threshold). You could do it by looking at the speed of the car when it disappeared and calculating whether it could have passed between readings, but that's more processing per second which does effect battery life on the Garmin.
